### PR TITLE
Added type-hints, reformatted arrays

### PR DESCRIPTION
**Pull request by Quix0r**
Changes:
- add type-hints
- replaced double-quotes with single
- changed a lot $array['foo'] into a single `$foo = ['foo' => $somethingFoo, 'bar' => $somethingBar];`

### CODE REVIEW

File: src/Network/Probe.php [hunk 1]
```diff
@@ -346,6 +346,7 @@ public static function lrdd(string $uri): array
 	 * @return array uri data
 	 * @throws HTTPException\InternalServerErrorException
 	 * @throws \ImagickException
+	 * @todo Choice: implement $cache parameter or remove documentation
```
Comment (MrPetovan) [inline src/Network/Probe.php]: Please remove the obsolete parameter documentation.

File: src/Protocol/Diaspora.php [hunk 2]
```diff
@@ -281,7 +282,7 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
 			if ($no_exit) {
 				return false;
 			} else {
-				throw new \Friendica\Network\HTTPException\BadRequestException();
+				throw new BadRequestException();
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
				throw new HTTPException\BadRequestException();
```

File: src/Protocol/Diaspora.php [hunk 3]
```diff
@@ -42,6 +42,7 @@
 use Friendica\Model\User;
 use Friendica\Network\HTTPClient\Client\HttpClientAccept;
 use Friendica\Network\HTTPException;
+use Friendica\Network\HTTPException\BadRequestException;
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
```

File: src/Protocol/Diaspora.php [hunk 4]
```diff
@@ -267,7 +268,7 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
 				if ($no_exit) {
 					return false;
 				} else {
-					throw new \Friendica\Network\HTTPException\BadRequestException();
+					throw new BadRequestException();
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
				throw new HTTPException\BadRequestException();
```

File: src/Protocol/Diaspora.php [hunk 5]
```diff
@@ -307,7 +308,7 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
 			if ($no_exit) {
 				return false;
 			} else {
-				throw new \Friendica\Network\HTTPException\BadRequestException();
+				throw new BadRequestException();
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
				throw new HTTPException\BadRequestException();
```

File: src/Protocol/Diaspora.php [hunk 6]
```diff
@@ -322,7 +323,7 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
...
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
				throw new HTTPException\BadRequestException();
```

File: src/Protocol/Diaspora.php [hunk 7]
```diff
@@ -332,7 +333,7 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
...
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: We already import `Friendica\Network\HTTPException`
```suggestion
				throw new HTTPException\BadRequestException();
```

File: src/Protocol/Diaspora.php [hunk 8]
```diff
@@ -2670,9 +2688,6 @@ private static function receiveStatusMessage(array $importer, SimpleXMLElement $
 		} elseif ($data->poll) {
 			$datarray['object-type'] = Activity\ObjectType::NOTE;
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: Overriding the key with the same value.
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by annando** [comment]
This PR causes the issue https://github.com/friendica/friendica/issues/13221 - this evening I will provide a fix.

**Comment by Quix0r** [comment]
> This PR causes the issue #13221 - this evening I will provide a fix.

Ops, sorry that this happened.

**Pull request merged**